pipert2/core/base/pipe.py

```diff
@@ -17,49 +20,62 @@
 add_pipe_log_level()
 
 
-class Pipe:
+class Pipe(BaseEventExecutor):
     """The pipe object is the center of the pipe.
     Once it is created it act as a central registry for the
     pipe methods such as create flows, register routines, notify events
     and more.
 
     """
 
-    def __init__(self, network: Network = QueueNetwork(),
+    events = class_functions_dictionary()
+
+    def __init__(self, event_board: EventBoard = EventBoard(), network: Network = QueueNetwork(),
                  logger: Logger = get_default_print_logger("Pipe"),
-                 data_transmitter: DataTransmitter = BasicTransmitter(),
-                 auto_pacing_mechanism: bool = False):
+                 data_transmitter: DataTransmitter = BasicTransmitter(), auto_pacing_mechanism: bool = False,
+                 run_rpc_cli: bool = False):
```

Comment:
I am sad to see that you chose to do this that way. buy i approve :disappointed: . (resolve when you see this)